pkgs/top-level/all-packages.nix

```diff
@@ -12566,6 +12566,16 @@ with pkgs;
     jdk = jdk8; # TODO: remove override https://github.com/NixOS/nixpkgs/pull/89731
   };
 
+  adoptopenjdk-bin-17-packages-linux = import ../development/compilers/adoptopenjdk-bin/jdk17-linux.nix { inherit lib; };
```

Comment:
What does the 'inherit lib' do here?

Reply:
This is cherry-picked from: https://github.com/NixOS/nixpkgs/pull/165354

afaik it is a common pattern on adoptopenjdk (check lines above/bellow following the same pattern) but I am not familiarized with why it is done that way.

Reply:
Yeah, I saw that but I couldn't explain it, so I was wondering if you could :). What happens if it's removed?